From 75df605ffca2120bfe49456ea2f8821fd2fe74f3 Mon Sep 17 00:00:00 2001
From: shucheng <shucheng@bd-apaas.com>
Date: Fri, 20 Sep 2024 12:54:24 +0800
Subject: [PATCH] lease env

---
 pkg/virt-operator/application.go | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/pkg/virt-operator/application.go b/pkg/virt-operator/application.go
index e2f50c9420..d78599ddf4 100644
--- a/pkg/virt-operator/application.go
+++ b/pkg/virt-operator/application.go
@@ -377,6 +377,11 @@ func (app *VirtOperatorApp) Run() {
 
 	leaseName := VirtOperator
 
+	// 读取环境变量并重载参数
+	if leaseNameEnv := os.Getenv("VIRT_OPERATOR_LEASE"); leaseNameEnv != "" {
+		leaseName = leaseNameEnv
+	}
+
 	recorder := app.getNewRecorder(k8sv1.NamespaceAll, leaseName)
 
 	id, err := os.Hostname()
-- 
2.39.2

